deeplearning4j-scaleout/spark/dl4j-spark-ml/src/main/scala/org/deeplearning4j/spark/ml/impl/SparkDl4jNetworkWrapper.scala

```diff
@@ -0,0 +1,117 @@
+package org.deeplearning4j.spark.ml.impl
+
+import java.util
+
+import org.apache.spark.ml.{PredictionModel, Predictor}
+import org.apache.spark.ml.param.ParamMap
+import org.apache.spark.ml.util._
+import org.apache.spark.mllib.linalg.Vector
+import org.apache.spark.mllib.regression.LabeledPoint
+import org.apache.spark.sql.Row
+import org.deeplearning4j.nn.conf.MultiLayerConfiguration
+import org.deeplearning4j.nn.multilayer.MultiLayerNetwork
+import org.deeplearning4j.optimize.api.IterationListener
+import org.deeplearning4j.spark.impl.multilayer.SparkDl4jMultiLayer
+import org.deeplearning4j.spark.ml.utils.{DatasetFacade, ParamSerializer}
+import org.deeplearning4j.util.ModelSerializer
+
+
+abstract class SparkDl4jNetworkWrapper[T, E <: SparkDl4jNetworkWrapper[T, E, M], M <: SparkDl4jModelWrapper[T, M]]
+(override val uid: String) extends Predictor[T, E, M]  {
+
+    protected var _multiLayerConfiguration : MultiLayerConfiguration = _
+    protected var _numLabels : Int = 2
+    protected var _freq : Int = 10
+    protected var _trainingMaster : ParamSerializer = _
+    protected var _listeners : util.Collection[IterationListener] = _
+    protected var _collectStats: Boolean = false
+
+    override def copy(extra: ParamMap) : E = defaultCopy(extra)
+
+    val mapVectorFunc : Row => LabeledPoint
+
+    def setNumLabels(value: Int) : E = {
+        this._numLabels = value
+        this.asInstanceOf[E]
+    }
+
+    def setMultiLayerConfiguration(multiLayerConfiguration: MultiLayerConfiguration) : E = {
```

Comment:
The abstract class is basically a wrapper to get around the spark 1 and spark 2 issue. I didn't want somebody to use this as the type when instantiating a network. In fact, it may be best to limit the scope of this class to protected.